[draft_export] Clear pending unbacked symbols when overriding mismatched fake kernels

yiming0416 opened the pull request:
Summary:
When encountering a mismatched fake kernel that also creates unbacked symbols, draft export will fail with `PendingUnbackedSymbolNotFound` error.

Clearing `shape_env.pending_fresh_unbacked_symbols` fixes this issue.

Test Plan:
```
buck2 run mode/dev-nosan caffe2/test:test_export -- -r test_override_mismatched_fake_kernel_with_unbacked_symbols
```

Differential Revision: D68920990

pianpwk reviewed (approved):
For more context, this happens when a meta-kernel allocates and uses unbacked symbols, but real-tensor tracing detects an incorrect implementation, and instead of erroring out, overrides the meta-kernel with an inferred one and continues. This clears out the pending unbacked symbols allocated by the user-implemented kernel.

@ezyang could you help verify if this is ok? I'm assuming the pending symbols should only come from the incorrect fake kernel, and we're not clearing out unbound symbols from previous operators?

yiming0416 committed:
[draft_export] Clear pending unbacked symbols when overriding mismatched fake kernels (#146089)

Summary:

When encountering a mismatched fake kernel that also creates unbacked symbols, draft export will fail with `PendingUnbackedSymbolNotFound` error.

Clearing `shape_env.pending_fresh_unbacked_symbols` fixes this issue.

Test Plan:
```
buck2 run mode/dev-nosan caffe2/test:test_export -- -r test_override_mismatched_fake_kernel_with_unbacked_symbols
```

Reviewed By: pianpwk

Differential Revision: D68920990
